src/main/java/io/camunda/zeebe/bpmnassert/testengine/IdleStateMonitor.java

```diff
@@ -0,0 +1,74 @@
+package io.camunda.zeebe.bpmnassert.testengine;
+
+import io.camunda.zeebe.engine.processing.streamprocessor.StreamProcessorListener;
+import io.camunda.zeebe.engine.processing.streamprocessor.TypedRecord;
+import io.camunda.zeebe.logstreams.log.LogStreamReader;
+import io.camunda.zeebe.logstreams.log.LoggedEvent;
+import io.camunda.zeebe.logstreams.storage.LogStorage;
+import java.util.ArrayList;
+import java.util.List;
+
+final class IdleStateMonitor implements LogStorage.CommitListener, StreamProcessorListener {
```

Comment:
❌ We should write some unit tests for this logic

Reply:
Unfortunately the `IdleStateMonitor` is not stable as it is now. I tried writing unit tests to verify the waiting for idle state works correctly, but they kept failing.
I propose we merge this PR first and have another look at this separate from this PR.